source/Octopus.Tentacle.Client/Scripts/ScriptServiceV2Orchestrator.cs

```diff
@@ -200,7 +200,10 @@ protected override async Task<ScriptStatusResponseV2> Finish(ScriptStatusRespons
             {
                 // Finish performs a best effort cleanup of the Workspace on Tentacle
                 // If we are cancelling script execution we abandon a call to complete script after a period of time
-                using var completeScriptCancellationTokenSource = new CancellationTokenSource(onCancellationAbandonCompleteScriptAfter);
+                
+                using var completeScriptCancellationTokenSource = new CancellationTokenSource();
+
+                await using var _ = scriptExecutionCancellationToken.Register(() => completeScriptCancellationTokenSource.CancelAfter(onCancellationAbandonCompleteScriptAfter));
```

Comment:
Shame a `.Register` has crept back in given it can easily become the source of unexpected bugs but it's the simplest way to do this and a simple usage of it!!

Alt could be Task.WhenAny with rpcTask and Task.Delay(-1, scriptExecutionCancellationToken wrapped in a cleanup source) but the orchestration logic and cleanup logic of cancellation tokens sources and tasks get a bit verbose and complex!!